internal/provisioner/rtcd.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package provisioner
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type rtcd struct {
+	awsClient      aws.AWS
+	environment    string
+	provisioner    *KopsProvisioner
+	kops           *kops.Cmd
+	cluster        *model.Cluster
+	logger         log.FieldLogger
+	desiredVersion *model.HelmUtilityVersion
+	actualVersion  *model.HelmUtilityVersion
+}
+
+func newRtcdHandle(cluster *model.Cluster, desiredVersion *model.HelmUtilityVersion, provisioner *KopsProvisioner, awsClient aws.AWS, kops *kops.Cmd, logger log.FieldLogger) (*rtcd, error) {
+	if logger == nil {
+		return nil, errors.New("cannot instantiate RTCD handle with nil logger")
+	}
+
+	if provisioner == nil {
+		return nil, errors.New("cannot create a connection to RTCD if the provisioner provided is nil")
+	}
+
+	if kops == nil {
+		return nil, errors.New("cannot create a connection to PgbRTCDouncer if the Kops command provided is nil")
+	}
+
+	return &rtcd{
+		awsClient:      awsClient,
+		environment:    awsClient.GetCloudEnvironmentName(),
+		provisioner:    provisioner,
+		kops:           kops,
+		cluster:        cluster,
+		logger:         logger.WithField("cluster-utility", model.RtcdCanonicalName),
+		desiredVersion: desiredVersion,
+		actualVersion:  cluster.UtilityMetadata.ActualVersions.Rtcd,
+	}, nil
+
+}
+
+func (p *rtcd) updateVersion(h *helmDeployment) error {
```

Comment:
Each of our methods refer to `(p *rtcd)` I believe this a copy paste error from pgbouncer, could we change them to be `(r *rtcd)` to prevent confusion?